gateway/src/apicast/cli/environment.lua

```diff
@@ -0,0 +1,129 @@
+local pl_path = require('pl.path')
+local resty_env = require('resty.env')
+local linked_list = require('apicast.linked_list')
+local setmetatable = setmetatable
+local loadfile = loadfile
+local pcall = pcall
+local require = require
+local assert = assert
+local error = error
+local print = print
+local pairs = pairs
+local ipairs = ipairs
+local tostring = tostring
+local insert = table.insert
+local concat = table.concat
+local re = require('ngx.re')
+
+local function parse_nameservers()
+    local resolver = require('resty.resolver')
+    local nameservers = {}
+
+    for _,nameserver in ipairs(resolver.init_nameservers()) do
+        insert(nameservers, tostring(nameserver))
+    end
+
+    if #nameservers > 0 then
```

Comment:
Why return nil instead of {} ?

Reply:
Because then it is easier to verify that there are no nameservers. `resolver` directive in the nginx conf has to have at least one server. So if there are none we can't set it.

Reply:
Would be good to document this.

Reply:
Hopefully https://github.com/3scale/apicast/pull/496/commits/5aa2e0c985aed253c5b29dba54a8a112d240e5c2 is clear enough and describes that the type of this can be a table of strings or nil. Would that be enough?